core/utils/cuckoo_map.h

```diff
@@ -180,7 +180,7 @@ class CuckooMap {
   Entry* Find(const K& key, const H& hasher = H(), const E& eq = E()) {
     // Blame Effective C++ for this
     return const_cast<Entry*>(
-        static_cast<const typeof(*this)&>(*this).Find(key, hasher, eq));
+        static_cast<const CuckooMap&>(*this).Find(key, hasher, eq));
```

Comment:
I found that `std::remove_reference` trick when I was looking at this last night. Does the same as the typeof.  Seems good.  I agree it's uglier, but oh well, that's C++11 for you!